packages/teleport/src/components/CatchError/CatchError.tsx

```diff
@@ -43,3 +57,18 @@ export default class CatchError extends React.Component {
     return this.props.children;
   }
 }
+
+type FallbackFnProp = {
+  error: Error;
+  retry(): void;
+};
+
+type State = {
+  error: Error;
+};
+
+type Props = {
+  children: React.ReactNode;
+  onRetry?(): void;
+  fallbackFn?(props: FallbackFnProp): React.ReactNode;
```

Comment:
Do we have a use case where we need fallbackFn, onRetry?

Reply:
yes sorry the usage is in my next PR but the usage looks like this:

```ts
    <CatchError
      onRetry={clearCachedJoinTokenResult}
      fallbackFn={props => (
        <Box>
          <Heading />
          <ScriptBox pollState="error">
            <TextIcon mt={2} mb={3}>
              <Icons.Warning ml={1} color="danger" />
              Encountered Error: {props.error.message}
            </TextIcon>
            <ButtonBlueText ml={2} onClick={props.retry}>
              Refetch a command
            </ButtonBlueText>
          </ScriptBox>
        </Box>
      )}
    >
      <Suspense
        fallback={
          <Box>
            <Heading />
            <ScriptBox height="144px">
              <Box textAlign="center" height="108px">
                <Indicator />
              </Box>
            </ScriptBox>
          </Box>
        }
      >
        <DownloadScript {...props} />
      </Suspense>
    </CatchError>
```